resources/python/update_requirements.py

```diff
@@ -221,7 +221,16 @@ def _update_requirements_file(self, dependencies):
                 # it to the right requirements file.
                 requirement_to_add = [dependency + "\n"]
                 if package_name in self._binary_distributions:
-                    bin_handler.writelines(requirement_to_add)
+                    if package_name == "cryptography":
+                        # Cryptography is a special use case for linux.
+                        # Default wheels requires GLIBC 2.28 (manylinux_2_28_x86_64) which is not available on CentOS 7.
+                        cryptography_to_add = [
+                             "cryptography==41.0.7 ; sys_platform != 'linux'\n",
+                             "https://files.pythonhosted.org/packages/14/fd/dd5bd6ab0d12476ebca579cbfd48d31bd90fa28fa257b209df585dcf62a0/cryptography-41.0.7-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl ; sys_platform == 'linux'\n"
+                        ]
```

Comment:
Looks good but could you move this test above the "# Figure which type of dependency it is and write" comment to make the code easier to read (in my opinion):

```python
for dependency in dependencies:
    package_name = dependency.split("==")[0]

    if package_name == "cryptography":
        # Cryptography is a special use case for linux.
        # Default wheels requires GLIBC 2.28 (manylinux_2_28_x86_64) which is not available on CentOS 7.
        dependency = "cryptography==41.0.7 ; sys_platform != 'linux'\n"
        dependency+= "https://files.pythonhosted.org/packages/14/fd/dd5bd6ab0d12476ebca579cbfd48d31bd90fa28fa257b209df585dcf62a0/cryptography-41.0.7-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl ; sys_platform == 'linux'"

    # Figure which type of dependency it is and write
    # it to the right requirements file.
    requirement_to_add = [dependency + "\n"]
    if package_name in self._binary_distributions:

...
```

I feel that, this way, the 2 different behaviors (whether the package is crypto, and whether the package is src or binary) are more separated and the overall function is easier to read.
I might be wrong.